### PR TITLE
Selected networks are not shown as preferred if a multichain wallet with preferred networks is scanned on the 'send to' page #19775

### LINKED ISSUE #19775: Selected networks are not shown as preferred if a multichain wallet with preferred networks is scanned on the 'send to' page
Follow up for https://github.com/status-im/status-mobile/pull/19668

### Steps:
1. `User A`sets up a multichain wallet QR with some preferred networks (e.g., 'arb1' only
image
)
2. `User B` goes to 'send to' page and scans `User A's` QR
3. `User B` Selects an asset for sending
4. `User B` enters the available asset amount on the asset sending page.
5. `User B` opens the 'edit receiver networks' drawer

### Actual result:
All networks are fetched for the receiver as preferable. This happens because, after scanning the QR code in step 2, the address is fetched without an 'arb1' prefix.

https://github.com/status-im/status-mobile/assets/52490791/d606aa98-ad92-49ee-ad33-f17a4efdc363

### Expected result:
The scanned addresses on the 'send to' page, should be fetched with networks prefixes. 

### OS: 
IOS, Android

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

### PR DESCRIPTION
**Pull request by mmilad75**
fixes #19775 

The issue was that the scanner was scanning the text successfully, but the filter we have in the `scan_account` was removing the prefixes of the address. I replaced it with the correct regex.

https://github.com/status-im/status-mobile/assets/55688834/a1606bd5-2fac-4ca0-b0c9-ee7ee25edf42

### CODE REVIEW

**Review by ilmotta — APPROVED** [review]
LGTM

Just one important note from my part, even though the regexes were extracted as constants, that doesn't give us any guarantee they are correct. Such complex regexes should definitely be covered by unit tests before we merge code IMO. This `regx-multichain-address` is not an easy one to grasp, but even if it was simpler I think it should be covered by tests. The macro `cljs.test/are` is our friend :)

**Comment by J-Son89** [comment]
would be nice to have a video if possible showing the behaviour here 🙏

**Review by J-Son89 — APPROVED** [review]

**Comment by mmilad75** [comment]
> would be nice to have a video if possible showing the behaviour here 🙏

Sure 👍

**Comment by mmilad75** [comment]
> would be nice to have a video if possible showing the behaviour here 🙏

Done

**Comment by VolodLytvynenko** [comment]
Hi @mmilad75 thank you for PR. Take a look at a found issue

## PR_ISSUE 1: 're-matches must match against the string' error is shown when an invalid eth QR is scanned

### Steps:
1. Scan QR which contains with invalid ETH address
image


### Actual result:
're-matches must match against the string' error is shown

https://github.com/status-im/status-mobile/assets/52490791/ec84a034-f49a-4e47-a2e9-209ba4f7ed8a

### Expected result:
No error is shown. If the invalid QR is scanned, then 'Oops this QR does not contain the address ' 
image

### OS: 
IOS, Android
### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

### Logs
[status_logs.zip](https://github.com/status-im/status-mobile/files/15222784/status_logs.zip)

**Comment by mmilad75** [comment]
> Hi @mmilad75 thank you for PR. Take a look at a found issue
> 
> ## PR_ISSUE 1: 're-matches must match against the string' error is shown when an invalid eth QR is scanned
> ### Steps:
> 1. Scan QR which contains with invalid ETH address
>    image
> 
> ### Actual result:
> 're-matches must match against the string' error is shown
> 
>  error_scan.mp4 
> ### Expected result:
> No error is shown. If the invalid QR is scanned, then 'Oops this QR does not contain the address ' image
> 
> ### OS:
> IOS, Android
> 
> ### Devices:
> * Pixel 7a, Android 13
> * iPhone 11 Pro Max, IOS 17
> 
> ### Logs
> [status_logs.zip](https://github.com/status-im/status-mobile/files/15222784/status_logs.zip)

Can you please have another look now?

**Comment by mmilad75** [comment]
> > Hi @mmilad75 thank you for PR. Take a look at a found issue
> > ## PR_ISSUE 1: 're-matches must match against the string' error is shown when an invalid eth QR is scanned
> > ### Steps:
> > 
> > 1. Scan QR which contains with invalid ETH address
> >    image
> > 
> > ### Actual result:
> > 're-matches must match against the string' error is shown
> > error_scan.mp4
> > ### Expected result:
> > No error is shown. If the invalid QR is scanned, then 'Oops this QR does not contain the address ' image
> > ### OS:
> > IOS, Android
> > ### Devices:
> > 
> > * Pixel 7a, Android 13
> > * iPhone 11 Pro Max, IOS 17
> > 
> > ### Logs
> > [status_logs.zip](https://github.com/status-im/status-mobile/files/15222784/status_logs.zip)
> 
> Can you please have another look now?

**Comment by mmilad75** [comment]
@VolodLytvynenko Can you please test again? It should be fine now

**Comment by VolodLytvynenko** [comment]
Discussing specification related to 'ethereum' prefix support https://discord.com/channels/624634427930312714/1225795996555022478/1237744788267073586

**Comment by mmilad75** [comment]
> Discussing specification related to 'ethereum' prefix support https://discord.com/channels/624634427930312714/1225795996555022478/1237744788267073586

This conversation is way beyond the scope of this ticket. Even if we want to support EIP-155 for the scanner, we need to add it to different part of the app (like when a user tries to add a EIP-155 address manually instead of scanning a qr-code). If we decide to add to support them, we need to create a separate ticket for it.

**Review by OmarBasem — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
> > Discussing specification related to 'ethereum' prefix support https://discord.com/channels/624634427930312714/1225795996555022478/1237744788267073586
> 
> This conversation is way beyond the scope of this ticket. Even if we want to support EIP-155 for the scanner, we need to add it to different part of the app (like when a user tries to add a EIP-155 address manually instead of scanning a qr-code). If we decide to add to support them, we need to create a separate ticket for it.

hi @mmilad75 Probably, we are going to use the approach as MetaMask does. Addresses that contain the 'ethereum' prefix will be recognized by our scanner, but only the address without the 'ethereum' prefix will be fetched into the 'Sent to' field, which helps the user to continue the sending flow.

### Steps:
1. Scan the address with `ethereum` prefix 
image


### Actual result in PR:
The address is not recognized by a scanner

https://github.com/status-im/status-mobile/assets/52490791/e800a702-e465-4e89-9cd0-7f5e055c925d



### How it works in develop and meta mask:
The address is recognized and fetched without 'ethereum' prefix

https://github.com/status-im/status-mobile/assets/52490791/a9c46878-bf6b-4651-8f5a-700b0fc0003d

This means that merging this PR can cause the breaking of the feature that supports fetching scanned addresses with the 'ethereum' prefix, which works now in develop.

Sorry for the confusion. I should have investigated and clarified this before creating this task:  https://github.com/status-im/status-mobile/issues/19775 task. Let's wait for John reply here https://discord.com/channels/624634427930312714/1225795996555022478/1237744788267073586

**Comment by VolodLytvynenko** [comment]
Hi @mmilad75, thank you for your work. The PR is ready to be merged. The next stage of address supporting will be addressed in the next PR: https://github.com/status-im/status-mobile/issues/19986

**Pull request merged**
